source/run/src/TG4PrimaryGeneratorAction.cxx

```diff
@@ -189,6 +192,140 @@ void TG4PrimaryGeneratorAction::TransformPrimaries(G4Event* event)
   }
 }
 
+
```

Comment:
There is a lot of code duplication wrt to previous method; could we introduce a helper function sahred by both methods?

Reply:
@ihrivnac can you check? I did introduce some helper code

Reply:
I checked the code now. Why do you need to save the particle position and momentum in the extra TMCParticleStatus object instead of using TParticle as what we do for transferring VMC stack primaries to G4 primaries?